parachain/runtime/tests/mock/issue_testing_utils.rs

```diff
@@ -0,0 +1,194 @@
+use crate::*;
+
+pub const USER: [u8; 32] = ALICE;
+pub const VAULT: [u8; 32] = BOB;
+pub const PROOF_SUBMITTER: [u8; 32] = CAROL;
+
+pub const DEFAULT_GRIEFING_COLLATERAL: u128 = 5_000;
+pub const DEFAULT_COLLATERAL: u128 = 1_000_000;
+
+pub const DEFAULT_USER_FREE_BALANCE: u128 = 1_000_000;
+pub const DEFAULT_USER_LOCKED_BALANCE: u128 = 100_000;
+pub const DEFAULT_USER_FREE_TOKENS: u128 = 1000;
+pub const DEFAULT_USER_LOCKED_TOKENS: u128 = 1000;
+
+pub fn request_issue(amount_btc: u128) -> (H256, IssueRequest<AccountId32, u32, u128, u128>) {
+    RequestIssueBuilder::new(amount_btc).request()
+}
+
+pub struct RequestIssueBuilder {
+    amount_btc: u128,
+    vault: [u8; 32],
+    user: [u8; 32],
+}
+impl RequestIssueBuilder {
+    pub fn new(amount_btc: u128) -> Self {
+        Self {
+            amount_btc,
+            vault: VAULT,
+            user: USER,
+        }
+    }
+    pub fn with_vault(&mut self, vault: [u8; 32]) -> &mut Self {
+        self.vault = vault;
+        self
+    }
+    pub fn with_user(&mut self, user: [u8; 32]) -> &mut Self {
+        self.user = user;
+        self
+    }
+    pub fn request(&self) -> (H256, IssueRequest<AccountId32, u32, u128, u128>) {
+        assert_ok!(ExchangeRateOracleModule::_set_exchange_rate(
+            FixedU128::one()
+        ));
+
+        SystemModule::set_block_number(1);
+
+        assert_ok!(Call::VaultRegistry(VaultRegistryCall::register_vault(
+            DEFAULT_COLLATERAL,
+            dummy_public_key()
+        ))
+        .dispatch(origin_of(account_of(self.vault))));
+
+        // alice requests polka_btc by locking btc with bob
+        assert_ok!(Call::Issue(IssueCall::request_issue(
+            self.amount_btc,
+            account_of(self.vault),
+            DEFAULT_GRIEFING_COLLATERAL
+        ))
+        .dispatch(origin_of(account_of(self.user))));
+
+        CollateralModule::transfer(
+            account_of(self.vault),
+            account_of(FAUCET),
+            CollateralModule::get_balance_from_account(&account_of(self.vault)),
+        )
+        .unwrap();
+
+        let issue_id = assert_issue_request_event();
+        let issue = IssueModule::get_issue_request_from_id(&issue_id).unwrap();
+
+        (issue_id, issue)
+    }
+}
+
+pub struct ExecuteIssueBuilder {
+    issue_id: H256,
+    issue: IssueRequest<AccountId32, u32, u128, u128>,
+    amount: u128,
+    submitter: [u8; 32],
+    relayer: [u8; 32],
+}
+impl ExecuteIssueBuilder {
```

Comment:
Very minor styling nit, I would prefer spacing between the `struct` and `impl`. Same for the following functions.

Reply:
Actually I may look into the rustfmt configuration...